docs/configure-rbac-permissions.md

```diff
@@ -0,0 +1,146 @@
+# Configure RBAC permissions
+
+`Microsoft.ApplicationInsights.Kubernetes` uses the default service account to query kubernetes information to enhance telemetries. It is important to have proper permissions configured for kubernetes related information like node, pod and so on to be fetched correctly.
+
+To help configure it successfully, in this post, we will start by describe a method to correctly configure the permissions for an RBAC enabled cluster. And then, we will share a guidance for troubleshooting.
+
+## Configure ClusterRole and ClusterRoleBinding for the service account
+
+* Create a yaml file, `sa-role.yaml` for example. We will deploy it when it is ready.
+
+* Write spec to define a cluster role, name it `appinsights-k8s-property-reader` for example:
+ 
+    ```yaml
+    kind: ClusterRole
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+    # "namespace" omitted since ClusterRoles are not namespaced
+    name: appinsights-k8s-property-reader
+    rules:
+    - apiGroups: ["", "apps"]
+    resources: ["pods", "nodes", "replicasets", "deployments"]
+    verbs: ["get", "list"]
+    ```
+    That spec defines the name of the role, and what permission does the role has, for example, list pods.
+
+    You don't have to use the exact name, but you will need to making sure the name is referenced correctly in the following steps.
+
+* Append a Cluster role binding spec, you will need to update the namespace accordingly:
+
+    ```yaml
+    ---
+    kind: ClusterRoleBinding
+    apiVersion: rbac.authorization.k8s.io/v1
+    metadata:
+    name: appinsights-k8s-property-reader-binding
+    subjects:
+    - kind: ServiceAccount
+    name: default
+    namespace: default
+    roleRef:
+    kind: ClusterRole
+    name: appinsights-k8s-property-reader
+    apiGroup: rbac.authorization.k8s.io
+    ```
+    
+    That is to grant the role of `appinsights-k8s-property-reader` to the default service account.
+    
+* Now you can deploy it:
+
+    ```shell
+    kubectl create -f `sa-role.yaml`
+    ```
+    See [sa-role.yaml](sa-role.yaml) for a full example.
+
+> :warning: Check back for various permissions needed. Depends on the implementations, it may change in the over time.
+
+## Ad-hoc troubleshooting for permission
+
+Kubectl provides an `auth --can-i` sub command for troubleshooting permissions. It supports impersonate the service account. We can leverage it for permission troubleshooting, for example:
+
+```shell
+kubectl auth can-i list pod --namespace default --as system:serviceaccount:default:default
+```
+
+There are several `default` there, let's break it down:
+
+* --namespace **default**: This is the target namespace, it is usually where your applications deploys to.
```

Comment:
I suggest using different namespace for the application (target namespace). Putting each application in a separate namespace is pretty much a standard practice in K8s world. It will make the instructions more realistic.

Reply:
I think it will make it more distinguishable too. Good point!